File: c/fibo.c
#include <stdio.h>
unsigned long long int fibo(unsigned long long int i, unsigned long long int j, unsigned int num){
	return num?fibo(j,i+j,num-1):i;
}

int main (void){
	unsigned int num;
	scanf("%u",&num);
	printf("%llu\n",fibo(0,1,num));
	return 0;
}

File: go/fibo/fibo.go
package main

import (
	"fmt"
)

func fibo(i, j, num uint64) uint64 {
	if num != 0 {
		return fibo(j, i+j, num-1)
	} else {
		return i
	}
}

func main() {
	var num uint64
	fmt.Scan(&num)
	fmt.Println(fibo(0, 1, num))
}
